src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -465,20 +486,46 @@ protected override ImmutableArray<PendingBranch> Scan(ref bool badRegion)
                 diagnostics,
                 useMethodSignatureParameterTypes: false,
                 methodSignatureOpt: null,
-                returnTypes: null,
                 initialState: null,
-                analyzedNullabilityMapOpt: null);
+                analyzedNullabilityMapOpt: null,
+                snapshotBuilderOpt: null,
+                returnTypesOpt: null);
         }
 
-        internal static void Analyze(
+        internal static ImmutableArray<(BoundReturnStatement, TypeWithAnnotations)> Analyze(
```

Comment:
>Analyze [](start = 84, length = 7)

Is this overload necessary, or could callers call into the other overload that takes an `ArrayBuilder` and avoid creating `ImmutableArray` instances?

Reply:
The only consumers of this overload use the produced `ImmutableArray`. I don't think it's a cost.

---
In reply to: [289515254](https://github.com/dotnet/roslyn/pull/35850#discussion_r289515254) [](ancestors = 289515254)

Reply:
The consumers previously used `ArrayBuilder`. Is there a reason to change that?

---
In reply to: [289538121](https://github.com/dotnet/roslyn/pull/35850#discussion_r289538121) [](ancestors = 289538121,289515254)